src/clib/pio_file.c

```diff
@@ -405,9 +417,9 @@ int PIOc_sync (int ncid)
   msg = PIO_MSG_SYNC;
 
   if(ios->async_interface && ! ios->ioproc){
-    if(ios->compmaster) 
+    if(!ios->comp_rank) 
```

Comment:
I don't see it here - did you push your change? (Push your change to the same branch to update the PR)